### PR TITLE
[move object type] Fix bad address function

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- `fn address` was incorrect for StakedSui
- Fixes issue with the transaction builder

## Test Plan 

- Added tests 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/dynamic_field.rs [hunk 1]
```diff
@@ -77,8 +82,17 @@ pub enum DynamicFieldType {
 impl DynamicFieldInfo {
     pub fn is_dynamic_field(tag: &StructTag) -> bool {
         tag.address == SUI_FRAMEWORK_ADDRESS
-            && tag.module.as_str() == "dynamic_field"
-            && tag.name.as_str() == "Field"
+            && tag.module.as_ident_str() == DYNAMIC_FIELD_MODULE_NAME
```
Comment (tnowacki) [inline crates/sui-types/src/dynamic_field.rs:85]: Not critical to this PR, but it made testing easier

File: crates/sui-types/src/dynamic_field.rs [hunk 2]
```diff
@@ -24,6 +26,9 @@ use shared_crypto::intent::HashingIntentScope;
 use std::fmt;
 use std::fmt::{Display, Formatter};
 
+const DYNAMIC_FIELD_MODULE_NAME: &IdentStr = ident_str!("dynamic_field");
+const DYNAMIC_FIELD_FIELD_STRUCT_NAME: &IdentStr = ident_str!("Field");
```
Comment (longbowlu) [inline crates/sui-types/src/dynamic_field.rs:30]: curious why `Field` is capitalized?

Comment (tnowacki) [reply crates/sui-types/src/dynamic_field.rs:30]: It's the name of the struct
https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/packages/sui-framework/sources/dynamic_field.move#L28

**Review by sblackshear — APPROVED** [review]

**Comment by sblackshear** [comment]
> and uncomment the thing I commented yesterday?

+1 to this

**Pull request merged**